packages/ckeditor5-paste-from-office/src/filters/table.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module paste-from-office/filters/parse
+ */
+import type { UpcastWriter, ViewDocumentFragment } from 'ckeditor5/src/engine';
+
+/**
+ * Set alignment for table pasted from MS Word.
+ *
+ * @param documentFragment The view structure to be transformed.
+ */
+export function tableAlignmentFilter( documentFragment: ViewDocumentFragment, writer: UpcastWriter ): void {
+	for ( const item of documentFragment.getChildren() ) {
+		// If table is not wrapped in div[align] it should be alligned left.
+		// More details:(https://github.com/ckeditor/ckeditor5/issues/8752#issuecomment-1623507171).
+		if ( item.is( 'element', 'table' ) ) {
+			writer.setAttribute( 'align', 'left', item );
+			continue;
+		}
+
+		const align = item.is( 'element' ) && item.getAttribute( 'align' );
+		const child = item.is( 'element' ) && item.getChild( 0 );
+
+		// Element should have alignment attribute and should have child and should be div.
+		if ( !align || !child || item.name !== 'div' ) {
+			continue;
+		}
+
+		// If table is wrapped in div[right|left] table should be alligned right or left.
+		// More details:(https://github.com/ckeditor/ckeditor5/issues/8752#issuecomment-1629065676)
+		if ( align !== 'center' && child.is( 'element', 'table' ) ) {
```

Comment:
Can't the following lines: [https://github.com/ckeditor/ckeditor5/blob/2b5758c590983b9bf06812c828662f060487a55f/packages/ckeditor5-paste-from-office/src/filters/table.ts#L33-L44](https://github.com/ckeditor/ckeditor5/blob/2b5758c590983b9bf06812c828662f060487a55f/packages/ckeditor5-paste-from-office/src/filters/table.ts#L33-L44)

Be simplified down to:

```javascript
if ( child.is( 'element', 'table' ) ) {
	writer.setAttribute( 'align', align === 'center' ? 'none' : align, child );
}
```

I don't have the option to run tests at hand.